src/Compilers/VisualBasic/Test/Semantic/Semantics/OverloadResolutionPriorityTests.vb

```diff
@@ -1560,5 +1562,1613 @@ End Module
+
+<C(""test"")>
+public class C
+    Inherits System.Attribute
+
+    <OverloadResolutionPriority(1)>
+    public Sub New(o As object)
+    End Sub
+
+    public Sub New(s As string)
+    End Sub
+End Class
+"
+
+            Dim compilation = CreateCompilation({source, OverloadResolutionPriorityAttributeDefinitionVB}, options:=TestOptions.DebugDll)
+            CompileAndVerify(compilation).VerifyDiagnostics()
+
+            Dim c = compilation.GetTypeByMetadataName("C")
+
+            Dim attr = c.GetAttributes().Single()
+            AssertEx.Equal("Sub C..ctor(o As System.Object)", attr.AttributeConstructor.ToTestDisplayString())
+        End Sub
+
+        <Fact>
+        Public Sub CycleOnOverloadResolutionPriorityConstructor_01()
+
+            Dim source = "
+namespace System.Runtime.CompilerServices
+    <AttributeUsage(AttributeTargets.Method Or AttributeTargets.Constructor Or AttributeTargets.Property, AllowMultiple:= false, Inherited:= false)>
+    public class OverloadResolutionPriorityAttribute
+        Inherits Attribute
+
+        <OverloadResolutionPriority(1)>
+        Public Sub New(priority As Integer)
+            Me.Priority = priority
+        End Sub
+
+        public Readonly Property Priority As Integer
+    End Class
+End Namespace
+"
+
+            Dim compilation = CreateCompilation(source, options:=TestOptions.DebugDll)
+            CompileAndVerify(compilation).VerifyDiagnostics()
+        End Sub
+
+        <Theory>
+        <InlineData(0)>
+        <InlineData(1)>
+        Public Sub CycleOnOverloadResolutionPriorityConstructor_02(ctorToForce As Integer)
+
+            Dim source = "
+Imports System.Runtime.CompilerServices
+
+namespace System.Runtime.CompilerServices
+    <AttributeUsage(AttributeTargets.Method Or AttributeTargets.Constructor Or AttributeTargets.Property, AllowMultiple:= false, Inherited:= false)>
+    public class OverloadResolutionPriorityAttribute
+        Inherits Attribute
+
+        Public Sub New(priority As Integer)
+            Me.Priority = priority
+        End Sub
+
+        ' Attribute is intentionally ignored, as this will cause a cycle
+        <OverloadResolutionPriority(1)>
+        Public Sub New(priority As Object)
+            Me.Priority = priority
+        End Sub
+
+        public Readonly Property Priority As Integer
+    End Class
+End Namespace
+
+public interface I1
+End Interface
+public interface I2
+End Interface
+public interface I3
+    Inherits I1, I2
+End Interface
+
+public Module Program 
+
+    <OverloadResolutionPriority(1)>
+    <Extension>
+    Sub M(this As I1)
+        System.Console.WriteLine(1)
+    End Sub
+
+    <Extension>
+    Sub M(this As I2)
+        throw DirectCast(Nothing, System.Exception)
+    End Sub
+
+    Sub Main
+        Dim i3 As I3 = Nothing
+        i3.M()
+    End Sub
+End Module
+"
+
+            Dim compilation = CreateCompilation(source, options:=TestOptions.DebugExe)
+            compilation.AssertTheseEmitDiagnostics(
```

Comment:
>compilation.AssertTheseEmitDiagnostics [](http://example.com/codeflow?start=12&length=38)

The equivalent C# test avoids verifying diagnostics, presumably to increase the chance of a cycle in attribute binding when using the semantic model. Is that not useful here?

Reply:
>Is that not useful here?

It is done below. The compilation is recreated after we check diagnostics